mbofui/app/js/controllers.js

```diff
@@ -1,112 +1,214 @@
+          type: 'xyz',
+        },
+        osmh: {
+          name: 'OpenStreetMap Hot',
+          type: 'xyz',
+          url: 'http://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png',
+        },
+        esriwsm: {
+          name: 'Esri World StreetMap',
+          type: 'xyz',
+          url: 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}'
+        },
+        esrisatellite: {
+          name: 'Esri Satellite',
+          type: 'xyz',
+          url: 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}'
         }
-
-        function posError(err) {
-            $log.info('ERROR(' + err.code + '): ' + err.message);
+      },
+      overlays: {
+        events: {
+          name: "Events",
+          type: "markercluster",
+          visible: true
+        },
+        draw: {
+          name: 'draw',
+          type: 'group',
+          visible: true,
+          layerParams: {
+            showOnSelector: false
+          }
         }
+      },
+      awesomeMarkerIcon: {
+        type: 'awesomeMarker',
+        markerColor: 'blue'
+      },
+    }
+  });
+
+  $scope.categories = null;
+  $scope.markers = [];
+  $scope.markersAll = [];
+
+
+  $scope.createEventCurrentlocation = function() {
+    leafletData.getMap().then(function(map) {
+      map.locate({
+        setView: true,
+        maxZoom: 16
+      });
+
+      function onLocationFound(e) {
+        var radius = e.accuracy / 2;
+        $('#bofModal').attr('data-coords', [e.latlng.lat, e.latlng.lng]);
+        L.popup()
+          .setLatLng(e.latlng)
+          .setContent('<a href="" data-toggle="modal" data-target="#bofModal">Add an event here?</a>')
+          .openOn(map);
+      }
+      map.on('locationfound', onLocationFound);
+    });
+  };
+
+  $scope.$on("leafletDirectiveMap.contextmenu", function(event, args) {
+    leafletData.getMap().then(function(map) {
+      var leafEvent = args.leafletEvent;
+      $('#bofModal').attr('data-coords', [leafEvent.latlng.lat, leafEvent.latlng.lng]);
+      L.popup()
+        .setLatLng(leafEvent.latlng)
+        .setContent('<a href="" data-toggle="modal" data-target="#bofModal">Add an event here?</a>')
+        .openOn(map);
+    });
+  });
+
+  $('#postBof').on('click', function() {
+    var coords = $('#bofModal').attr('data-coords').split(',');
+    var url = '/api/messages/';
+    var startTime = moment($scope.newStartTime).format('YYYY-MM-DDTHH:mm:ssZ');
+    var endTime = moment($scope.newEndTime).format('YYYY-MM-DDTHH:mm:ssZ');
+    var postingTime = moment().format('YYYY-MM-DDTHH:mm:ssZ');
+    var data = {
+      "messageText": $scope.newMessageText,
+      "startTime": startTime,
+      "endTime": endTime,
+      "latitude": coords[0],
+      "longitude": coords[1],
+      "altitudeMeters": 266.75274658203125, //this is being removed but the database still expects it
```

Comment:
I would say that the altitude should be kept.  It may not be completely accurate for now, but if clients can provide it, it doesn't take up much space and when accuracy improves in the future, you'll already have it.

The altitude may be useful for forensic purposes, right?  😉